File: ChatWebApp/Entities/Message.cs
namespace ChatAppAPI.Entities
{
    public class Message : IAuditable
    {
        public Guid Id { get; set; }
        public string Content { get; set; }
        public Guid ConversationParticipantId { get; set; }
        public ConversationParticipant ConversationParticipant{ get; set; }

        public DateTime CreatedDate { get; set; }
        public DateTime ModifiedDate { get; set; }
        public User? CreatedBy { get; set; }
        public User? ModifiedBy { get; set; }
    }
}

File: ChatWebApp/ClientApp/src/app/_models/message.ts
import { UserProfile } from "./user";

export class Message {
  id?: string;
  content?: string;
  conversationParticipantId?: string;
  createdDate?: Date;
  modifiedDate?: Date;
  createdBy?: UserProfile;
}

export class FirstMessageForCreation
{
  content?: string;
  name?: string;
  participants?: string[];
  avatar?: string;
  sender?: string;
}
